contracts/LivepeerProtocol.sol

```diff
@@ -1,756 +1,28 @@
-
-        return block.number >= rewardTimeWindowStartBlock && block.number < rewardTimeWindowEndBlock;
-    }
-
-    /*
-     * Returns total bonded stake for a current active transcoder
-     * @param _transcoder Address of a transcoder
-     */
-    function currentActiveTranscoderTotalStake(address _transcoder) constant returns (uint256) {
-        // Check if current active transcoder
-        if (!isCurrentActiveTranscoder[_transcoder]) throw;
-
-        return currentActiveTranscoders[currentActiveTranscoderPositions[_transcoder]].key;
-    }
-
-    /*
-     * Returns total bonded stake for a transcoder
-     * @param _transcoder Address of transcoder
-     */
-    function transcoderTotalStake(address _transcoder) constant returns (uint256) {
-        return transcoderPools.transcoderStake(_transcoder);
-    }
-
-    /*
-     * Returns bonded stake for a delegator. Accounts for rewards since last state transition
-     * @param _delegator Address of delegator
-     */
-    function delegatorStake(address _delegator) constant returns (uint256) {
-        // Check for valid delegator
-        if (!delegators[_delegator].initialized) throw;
-
-        return delegators[_delegator].bondedAmount.add(delegatorRewards(_delegator));
-    }
-
-    /*
-     * Computes rewards for a delegator since last state transition
-     * @param _delegator Address of delegator
-     */
-    function delegatorRewards(address _delegator) internal constant returns (uint256) {
-        uint256 rewards = 0;
-
-        // Check if delegator bonded to a transcoder
-        if (delegators[_delegator].transcoderAddress != address(0)) {
-            // Iterate from round that delegator last transitioned states to current round
-            // If the delegator is bonded to a transcoder, it has been bonded to the transcoder since lastStateTransitionRound
-            for (uint256 i = delegators[_delegator].lastStateTransitionRound; i <= currentRound(); i++) {
-                uint256[2] rewardMultiplier = rewardMultiplierPerTranscoderAndRound[delegators[_delegator].transcoderAddress][i];
-
-                // Check if transcoder has a reward multiplier for this round (total minted tokens for round > 0)
-                if (rewardMultiplier[0] > 0) {
-                    // Calculate delegator's share of reward
-                    uint256 delegatorShare = rewardMultiplier[0].mul(delegators[_delegator].bondedAmount).div(rewardMultiplier[1]);
+    // CONTROLLER ADMIN
 
-                    rewards = rewards.add(delegatorShare);
-                }
-            }
-        }
+    function updateController(bytes32 _key, address _controller) onlyOwner public returns (bool) {
+        // Check if key is in registry
+        if (!registryContains(_key)) throw;
 
-        return rewards;
+        return registryGet(_key).call(bytes4(keccak256("setController(address)")), _controller);
```

Comment:
As this line is a little hairy, I think it at least warrants a comment of why we're using the `.call()` indirection through the registry.